src/test_dll.py

```diff
@@ -0,0 +1,167 @@
+"""Test double linked list class."""
+import pytest
+
+
+@pytest.fixture
+def dll():
+    """Instantiate a dll for testing."""
+    from dll import DoubleLinkedList
+    double_link = DoubleLinkedList()
+    return double_link
+
+
+def test_dll_initialization(dll):
+    """Test that there's nothing initialized."""
+    assert dll.tail is None
+    assert dll.head is None
+
+
+def test_dll_push_raises_type_error(dll):
+    """Test that the value error  is raised."""
+    with pytest.raises(TypeError):
+        dll.push()
```

Comment:
This doesn't need to be tested. If you don't pass a value into a function that requires a value, Python raises the TypeError for you. This is just testing that functions work in general the way that they're supposed to work.